omnigan/data.py

```diff
@@ -149,25 +144,24 @@ def __getitem__(self, i):
         """
         paths = self.samples_paths[i]
 
-        
-        if self.transform:
-            return {
-                "data": self.transform(
-                    {task: pil_image_loader(path, task) for task, path in paths.items()}
-                ),
-                "paths": paths,
-                "domain": self.domain,
-                "mode": self.mode,
-            }
-
-        return {
-            "data": {
-                    task: pil_image_loader(path, task) for task, path in paths.items()
-            },
+        # always apply transforms,
+        # if no transform is specified, ToTensor and Normalize will be applied
+
+        item = {
+            "data": self.transform(
+                {
+                    task: pil_image_loader(path, task, self.domain)
+                    for task, path in paths.items()
+                }
+            ),
             "paths": paths,
             "domain": self.domain,
             "mode": self.mode,
         }
+        if "d" in item["data"]:
```

Comment:
the issue with having it out of `transform` is that you will get non-matching inputs: if the input is flipped, then the depth map should be too, if there's random cropping, it should be the same for all data etc.

Reply:
Yes that’s exactly the reason why I put it back here. It’s because transforms are applied on Images and so the depth image will be flipped cropped etc exactly as the others. This if call only gets the transformed depth image to converted depth tensor format

Reply:
There shouldn’t be any mismatch problem

Reply:
oooouh ok thanks :)